### DISCUSSION TITLE
How can i use MessageBox, Snackbar and Dialog

### DISCUSSION THREAD

**Discussion by asenyurt** [lepoco/wpfui#585] [Q&A]
How can i use MessageBox, Snackbar and Dialog ??

And is it possible to enter input textbox in Dialog ??

**Comment by ultravelocity** [comment]
Inject the service (e.g., ISnackbarService) into your viewmodel or page, and then call the relevant method (e.g., Show).

**Comment by wushuaixue** [comment]
I actually dont know how to use these, but there is a way to use it easily in your application.
```c#
var messageBox = new Wpf.Ui.Controls.MessageBox();
messageBox.Title = "Save your work?";
messageBox.Content = "Save your Project and all your setting?";
messageBox.CloseButtonText = "Cancel";
messageBox.SecondaryButtonText = "Don't Save";
messageBox.PrimaryButtonText =  "Save";
var result =  messageBox.ShowDialogAsync();
```

**Comment by Mithicor** [comment]
This is what I use for MessageBoxes so they are reusable.


The first one is if you want to return a result, such as if the user clicks yes or no.
```
public static async Task<Wpf.Ui.Controls.MessageBoxResult> GetMessageBoxResult(string title, string content, string primaryButtonText, string closeButtonText)
{
    var uiMessageBox = new Wpf.Ui.Controls.MessageBox
    {
        Title = title,
        Content = content,
        PrimaryButtonText = primaryButtonText,
        CloseButtonText = closeButtonText
    };

    return await uiMessageBox.ShowDialogAsync();
}
```


The Second is more just informational so all they can really do is hit ok to close it.
```
public static async Task ShowMessageBox(string title, string content, string closeButtonText)
{
    var uiMessageBox = new Wpf.Ui.Controls.MessageBox
    {
        Title = title,
        Content = content,
        CloseButtonText = closeButtonText
    };

    await uiMessageBox.ShowDialogAsync();
}
```


Refer to this page for info on Dialogs: [Dialog Info](https://github.com/lepoco/wpfui/discussions/848#discussioncomment-7711157)



Assuming you have your snackbar set up, you can "trigger" it like so:

```
snackbarService.Show("Title",
                      "Description",
                      ControlAppearance.Success,
                      new SymbolIcon(SymbolRegular.Fluent24), TimeSpan.FromSeconds(3));
```


I am not currently using injection in my projects, so here is how you can set up snackbars and dialogs without it:

In your Window that you want to display said snackbar or dialog, you want to put these:

```
<Window x:Class="MainWindow" ...>
  <Grid>
    // YOUR OTHER CONTROLS AND CONTENT GOES HERE

    <ui:SnackbarPresenter  x:Name="SnackbarPresenter"/>
    <ContentPresenter x:Name="RootContentDialog"/>
  </Grid>
```

Wherever you are keeping your properties (be it the window or a Viewmodel) create the service properties. You will set the presenters that you placed in your window XAML from the Window Constructor. The example below is not using a ViewModel:

```
public partial class MainWindow
{
  public ISnackbarService snackbarService { get; set; }
  public IContentDialogService dialogService { get; set; }
  
  public MainWindow()
  {
      InitializeComponent();

      dialogService = new ContentDialogService();
      snackbarService = new SnackbarService();

      dialogService.SetDialogHost(RootContentDialog);
      snackbarService.SetSnackbarPresenter(SnackbarPresenter);
  }
}
```
